tests/support_server/setup.pm

```diff
@@ -406,6 +408,35 @@ sub setup_iscsi_server {
     $iscsi_server_set = 1;
 }
 
+sub setup_iscsi_tgt_server {
+    return if $iscsi_tgt_server_set;
+
+    systemctl 'start tgtd';
+
+    # Configure default iscsi iqn
+    my $iqn = get_var "ISCSI_IQN", "iqn.2016-02.de.openqa";
+
+    # Get device for iscsi export
+    my $device = script_output "ls /dev/[sv]db";
+    die "detection of disk for iSCSI LUN failed" unless $device;
+
+    # Create new iqn target with target id 1
+    tgt_new_target(1, $iqn);
+    # Add device lun 1 to taget with id 1
+    tgt_new_lun(1, 1, "$device");
+    # Add device lun 2 to target with id 1 in case of multipath
+    tgt_new_lun(1, 2, "$device") if get_var('ISCSI_MULTIPATH');
+    # Set iscsi_id for lun 1
+    tgt_update_lun(1, 1, "iscsi-path-1");
+    # Set iscsi_id for lun 2 in case of multipath
+    tgt_update_lun(1, 2, "iscsi-path-1") if get_var('ISCSI_MULTIPATH');
+    # Authorize all clients
+    tgt_auth_all(1);
+    # Show details about configured iscsi server
+    tgt_show;
+    $iscsi_tgt_server_set = 1;
+}
+
```

Comment:
It seems we are nearing the limit of maintainability for the support-server …

Reply:
I am not an active user of the support server so I do not know that well. Creating a WIP PR and inviting the last contributors to the support server to the review would be cool, thx

Reply:
@thehejik , @ldevulder - WDYT ?

Reply:
I got simple idea and it can be checked in this WIP PR: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/6065

Reply:
Please do not invent a wheel again, instead of that just extend current SUPPORT_SERVER_ROLES with "iscsi" value in setup.pm and you are done.

I don't think that code is unmaintainable, with a new modules approach it would be even worst IMO than the current approach is. The code is not so complicated as we have only few possible roles still.

Reply:
I can confirm that debugging of enormous script is tricky when support server has multiple roles. On top we could greatly improve mutexes logic, as right now any service specific mutex is not any better than `support_server_ready` as script for all services is executed at once. With smaller intermediate steps we could really start using e.g. pxe setup before dhcp is configured, etc.

Reply:
@ldevulder yes, approach with tgtadm is compatible with virtio console (no needles, fast execution), but functionality needed for HA iscsi would have to be added there. If you want,  we can create progress issue and add HA setup requirements for HA tests (and remove that yast iscsi config).

Reply:
Yes, I think I will do this in a near future.